src/core/application/useCases/pedido/index.ts

```diff
@@ -1,15 +1,55 @@
+import { Pedido, StatusPedidoEnum } from "@domain/entities/pedido";
 import { IPedidoUseCase } from "./pedido.interface";
-import { Pedido } from "@domain/entities/pedido";
 import { AssertionConcern } from "@domain/base/assertionConcern";
-import { IPedidoRepository } from "@domain/repositories/pedidoRepository.interface";
+import { PedidoRepository } from "@domain/repositories/pedidoRepository.interface";
+import { ProdutoRepository } from "@domain/repositories/produtoRepository.interface";
+import { ValorTotal } from "@domain/valueObjects/valorTotal";
 
 export class PedidoUseCase implements IPedidoUseCase {
-    constructor(private readonly pedidoRepository: IPedidoRepository) {}
+    private readonly pedidoRepository: PedidoRepository;
+    private readonly produtoRepository: ProdutoRepository;
+
+    constructor(
+        pedidoRepository: PedidoRepository,
+        produtoRepository: ProdutoRepository,
+    ) {
+        this.pedidoRepository = pedidoRepository;
+        this.produtoRepository = produtoRepository;
+    }
 
     getAll(filters?: Partial<Pedido>): Promise<Pedido[]> {
```

Comment:
Mesma sugestão aqui pra deixar com o public async

```suggestion
    public async getAll(filters?: Partial<Pedido>): Promise<Pedido[]> {
```